### PR TITLE
Update async.c to handle segfaults for async connections

### PR DESCRIPTION
**Pull request by thisisbasil**
fixing the amateur hour redis free code

### CODE REVIEW

File: async.c [hunk 1]
```diff
@@ -360,6 +360,8 @@ static void __redisRunDisconnectCallback(redisAsyncContext *ac, int status)
 
 /* Helper function to free the context. */
 static void __redisAsyncFree(redisAsyncContext *ac) {
+    if (!ac) return;
```
Comment (zeekling) [inline async.c:363]: When the pointer ac is NULL?

**Comment by michael-grunder** [comment]
> fixing the amateur hour redis free code

I'm not inclined to merge this PR with that commit message.  The fact that it's obnoxious aside, it's useless in a `git log` as it doesn't say what it's actually doing.

See 21a1207ae13f3d422fb99669019836da3422b784 for a better example.

I'm tagging v1.1.0-rc1 tonight, so feel free to resubmit with a helpful commit message and I will get it merged and into the release.

File: async.c [hunk 2]
```diff
@@ -413,6 +415,8 @@ static void __redisAsyncFree(redisAsyncContext *ac) {
  * free'ing. To do so, a flag is set on the context which is picked up by
  * redisProcessCallbacks(). Otherwise, the context is immediately free'd. */
 void redisAsyncFree(redisAsyncContext *ac) {
+    if (!ac) return;
+
```
Comment (michael-grunder) [inline async.c:419]: Please remove whitespace.

File: async.c [hunk 3]
```diff
@@ -360,6 +360,8 @@ static void __redisRunDisconnectCallback(redisAsyncContext *ac, int status)
 
 /* Helper function to free the context. */
 static void __redisAsyncFree(redisAsyncContext *ac) {
+    if (!ac) return;
+
```
Comment (michael-grunder) [inline async.c:364]: Whitespace.

**Comment by michael-grunder** [comment]
Closing in favor of #1135

**Comment by thisisbasil** [comment]
sorry for the tone, this issue caused a problem with production code. good to see it is implemented

**Comment by michael-grunder** [comment]
No worries, we've all been there!

**Pull request closed without merging**